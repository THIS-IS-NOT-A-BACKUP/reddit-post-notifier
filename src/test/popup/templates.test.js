import { JSDOM } from 'jsdom';
import './mocks/browser.mock';
import getTemplates from '../../scripts/popup/templates';
import html from '../../common/popup.html';

beforeAll(() => {
    window.document.body.innerHTML = new JSDOM(html).window.document.body.innerHTML;
});

describe('templates', () => {
    test('should match snapshot', () => {
        expect(getTemplates()).toMatchSnapshot();
    });
});
