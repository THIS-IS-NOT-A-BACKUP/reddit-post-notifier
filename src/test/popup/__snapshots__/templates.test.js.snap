// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates should match snapshot 1`] = `
Object {
  "noPosts": <DocumentFragment>
    
                
    <div
      class="no-posts"
    >
      No unread posts
    </div>
    
            
  </DocumentFragment>,
  "postList": <DocumentFragment>
    
                
    <ul
      class="post-list"
    />
    
            
  </DocumentFragment>,
  "postListRow": <DocumentFragment>
    
                
    <li
      tabindex="1"
    >
      
                    
      <button
        class="check-mark"
        data-l10n-attrs="title"
        data-l10n-id="postListCheckMark"
        data-l10n-nocontent="true"
      >
        
                        
        <svg
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          
                            
          <use
            href="#check-mark-path"
          />
          
                        
        </svg>
        
                    
      </button>
      
                    
      <a
        href="#"
      />
      
                
    </li>
    
            
  </DocumentFragment>,
  "queryList": <DocumentFragment>
    
                
    <ul
      class="query-list"
    />
    
            
  </DocumentFragment>,
  "queryListRow": <DocumentFragment>
    
                
    <li
      tabindex="1"
    >
      
                    
      <button
        class="check-mark"
        data-l10n-attrs="title"
        data-l10n-id="queryListCheckMark"
        data-l10n-nocontent="true"
      >
        
                        
        <svg
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          
                            
          <use
            href="#check-mark-path"
          />
          
                        
        </svg>
        
                    
      </button>
      
                    
      <div
        class="query-name"
      />
      
                    
      <button
        class="arrow-right"
      >
        
                        
        <svg
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          
                            
          <path
            clip-rule="evenodd"
            d="M8.293 4.293a1 1 0 0 1 1.414 0l7 7a1 1 0 0 1 0 1.414l-7 7a1 1 0 0 1-1.414-1.414L14.586 12 8.293 5.707a1 1 0 0 1 0-1.414z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          
                        
        </svg>
        
                    
      </button>
      
                
    </li>
    
            
  </DocumentFragment>,
}
`;
